### PR TITLE
doc: add Docker credential store and environment variable configuration

### PR DESCRIPTION
**Pull request by FeynmanZhou**
Fix https://github.com/notaryproject/notaryproject.dev/issues/116. This is still a frequently asked question, such as https://github.com/notaryproject/notaryproject/issues/251

### CODE REVIEW

**Comment by FeynmanZhou** [comment]
@zr-msft  Could you pls help to review it? Thanks

File: content/en/docs/how-to/registry-authentication.md [hunk 1]
```diff
@@ -17,7 +17,67 @@ To authenticate to an OCI-compliant registry, use the `notation login` command w
...
+Then you will be able to interact with the registry and sign artifacts using Notation.
+
+For security reasons, you would better remove the existing environment variables after you log out from the registry.
+
+```console
+unset NOTATION_USERNAME
+NOTATION_PASSWORD
+```
+
+## Configure Docker Credential Store
```
Comment (zr-msft) [inline content/en/docs/how-to/registry-authentication.md:40]:
```suggestion
## Configure Docker Credential Store for Linux
```

File: content/en/docs/how-to/registry-authentication.md [hunk 2]
```diff
@@ -17,7 +17,67 @@ To authenticate to an OCI-compliant registry, use the `notation login` command w
...
+
+As local credentials may be required to access the remote registries, they need to be stored and accessed securely. To achieve maximum security, credential helpers are preferred so that credentials are stored in the system key chain with better protection. 
+
+To achieve maximum compatibility with existing systems, [Docker Credential Helpers](https://github.com/docker/docker-credential-helpers) and its [protocol](https://docs.docker.com/engine/reference/commandline/login/#credential-helper-protocol) are adopted as the credential helpers for Notation. Currently, it requires manual installation and configuration in Linux by following the steps below. This configuration will be simplified in Notation v1.0.0 with [oras-credential-go](https://github.com/oras-project/oras-credentials-go/discussions/18) support.
```
Comment (zr-msft) [inline content/en/docs/how-to/registry-authentication.md:44]:
```suggestion
`notation` uses [Docker Credential Helpers](https://github.com/docker/docker-credential-helpers) and its [protocol](https://docs.docker.com/engine/reference/commandline/login/#credential-helper-protocol) as the credential helpers. Currently, using Docker Credential Helpers requires manual installation and configuration in Linux by following the steps below. This configuration will be simplified in Notation v1.0.0[oras-credential-go](https://github.com/oras-project/oras-credentials-go/discussions/18).
```

File: content/en/docs/how-to/registry-authentication.md [hunk 3]
```diff
@@ -17,7 +17,67 @@ To authenticate to an OCI-compliant registry, use the `notation login` command w
...
+
+Install [Docker credential helper](https://github.com/docker/docker-credential-helpers) pass.
+
+```console
+mkdir -p ~/bin
+curl -Lo ~/bin/docker-credential-pass "https://github.com/docker/docker-credential-helpers/releases/download/v0.7.0/docker-credential-pass-v0.7.0.linux-amd64"
+chmod +x ~/bin/docker-credential-pass
+```
+
+Configure GPG key for encryption. If you have a GPP key already, run `gpg --edit-key` to trust your key.
```
Comment (zr-msft) [inline content/en/docs/how-to/registry-authentication.md:54]:
```suggestion
Configure GPG key for encryption. 

> **NOTE:** If you have a GPP key already, run `gpg --edit-key` to trust your key instead.
```

**Review by zr-msft — CHANGES_REQUESTED** [review]
Some minor cleanup, style, and consolidation

File: content/en/docs/how-to/registry-authentication.md (hunk 3 above)
Comment (FeynmanZhou) [reply content/en/docs/how-to/registry-authentication.md]: Updated

**Comment by FeynmanZhou** [comment]
> Some minor cleanup, style, and consolidation

@zr-msft Thanks! I updated it with the changes you suggested.

File: content/en/docs/how-to/registry-authentication.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ weight: 1
 
 You can use `notation` to authenticate to an OCI-compliant registry. This is useful for pushing and pulling signed artifacts, as well as inspecting artifacts in registries that do not offer public access.
 
-> **Important** Currently, `notation` relies on Docker Credential Store for authentication, which is included in Docker Desktop. For more information, see [Docker Credential Store](https://github.com/notaryproject/notation/blob/main/specs/registry-auth.md#credential-store)
+> **Important** Currently, `notation` relies on Docker Credential Store for authentication, which is included in Docker Desktop. It requires additional configuration if you using Notation in Linux. For more information, see [Docker Credential Store](https://github.com/notaryproject/notation/blob/main/specs/registry-auth.md#credential-store).
```
Comment (priteshbandi) [inline content/en/docs/how-to/registry-authentication.md:10]:
```suggestion
> **Important** Currently, `notation` relies on Docker Credential Store for authentication, which is included in Docker Desktop. It requires additional configuration if you using Notation without docker desktop, for more information, see [Docker Credential Store](https://github.com/notaryproject/notation/blob/main/specs/registry-auth.md#credential-store).
```

File: content/en/docs/how-to/registry-authentication.md [hunk 5]
```diff
@@ -17,7 +17,69 @@ To authenticate to an OCI-compliant registry, use the `notation login` command w
...
+## Configure environment variables to authenticate to an OCI-compliant registry
+
+Set the `NOTATION_USERNAME` and `NOTATION_PASSWORD` environment variables to authenticate to an OCI-compliant registry.
+
+```console
+export NOTATION_USERNAME="YOUR_REGISTRY_USERNAME"
+export NOTATION_PASSWORD="YOUR_REGISTRY_PASSWORD"
+```
+
+After the environment variables are set, you can use `notation login` without specifying a username or password.
```
Comment (priteshbandi) [inline content/en/docs/how-to/registry-authentication.md:31]: user doesn't need to use notation login after env variable.

Comment (shizhMSFT) [reply content/en/docs/how-to/registry-authentication.md:31]: +1

File: content/en/docs/how-to/registry-authentication.md [hunk 6]
```diff
@@ -17,7 +17,69 @@ To authenticate to an OCI-compliant registry, use the `notation login` command w
...
 ```
 
-> **Note:** If `notation login` is failing, you may need to configure [Docker Credential Store](https://github.com/notaryproject/notation/blob/main/specs/registry-auth.md#credential-store) for your environment.
+> **Note:** If `notation login` is failing, you may need to configure [Docker Credential Store](https://github.com/notaryproject/notation/blob/main/specs/registry-auth.md#credential-store) for your environment or configure environment variables as detailed in the *Configure Docker Credential Store for Linux* section.
+
+## Configure environment variables to authenticate to an OCI-compliant registry
+
+Set the `NOTATION_USERNAME` and `NOTATION_PASSWORD` environment variables to authenticate to an OCI-compliant registry.
+
+```console
```
Comment (shizhMSFT) [inline content/en/docs/how-to/registry-authentication.md:26]:
```suggestion
```shell
```
or
```suggestion
```bash
```
will be better

File: content/en/docs/how-to/registry-authentication.md [hunk 7]
```diff
@@ -17,7 +17,69 @@ To authenticate to an OCI-compliant registry, use the `notation login` command w
...
+
+Install and configure [pass](https://www.passwordstore.org/).
+
+```console
+sudo apt update 
+sudo apt install pass -y
+pass init $your_email
+```
+
+Configure Docker credential store in `/.docker/config.json`.
```
Comment (shizhMSFT) [inline content/en/docs/how-to/registry-authentication.md:70]:
```suggestion
Configure Docker credential store in `~/.docker/config.json`.
```

File: content/en/docs/how-to/registry-authentication.md [hunk 8]
```diff
@@ -17,7 +17,69 @@ To authenticate to an OCI-compliant registry, use the `notation login` command w
...
+```
+mkdir -p ~/.docker
+echo '{"credsStore":"pass"}' > ~/.docker/config.json
+chmod 600 ~/.docker/config.json
+```
+
+Use `notation login` to authenticate to an OCI-compliant registry.
+
+```console
+notation login -u <username> -p <password> <registry>
```
Comment (shizhMSFT) [inline content/en/docs/how-to/registry-authentication.md:81]: Passwords should be typed via stdin, and should not be passed as a command line argument.
```suggestion
notation login -u <username> <registry>
```
`-p` is there just for dev / testing purposes.

The security behind it is that any executed shell command might be logged somewhere in plaintext.

File: content/en/docs/how-to/registry-authentication.md (hunk 4 above)
Comment (iamsamirzon) [reply content/en/docs/how-to/registry-authentication.md:10]: Suggest re-write the 2nd sentence to make it clearer from 
~It requires additional configuration if you using Notation without docker desktop, for more information ...~
Notation requires additional configuration when used without docker desktop, for more information ...

**Review by iamsamirzon — DISMISSED** [review]

File: content/en/docs/how-to/registry-authentication.md (hunk 8 above)
Comment (FeynmanZhou) [reply content/en/docs/how-to/registry-authentication.md:81]: Agree. I realized that the login prompt feature is supported in RC.4.

File: content/en/docs/how-to/registry-authentication.md (hunk 5 above)
Comment (FeynmanZhou) [reply content/en/docs/how-to/registry-authentication.md]: Updated

File: content/en/docs/how-to/registry-authentication.md (hunk 4 above)
Comment (FeynmanZhou) [reply content/en/docs/how-to/registry-authentication.md]: I find that Docker Desktop relies on a Credential Helper [pass](https://www.passwordstore.org/) to store credentials on Linux. Linux users still have some manual configuration steps as demonstrated at [Credentials management for Linux users](https://docs.docker.com/desktop/get-started/#credentials-management-for-linux-users). So we need to emphasize this configuration is used for Linux.

**Review by iamsamirzon — APPROVED** [review]

**Review by vaninrao10 — APPROVED** [review]
LGTM

**Review by yizha1 — APPROVED** [review]
LGTM

**Pull request merged**